### DISCUSSION TITLE
Hold and Release Actions on Hardware Buttons

### DISCUSSION THREAD

**Discussion by irvintim** [Blackymas/NSPanel_HA_Blueprint#1854] [Q&A]
I hope this is possible; I'm just unsure how to do it.

My ceiling light has 2 entities for dimming:
- `button.bedroom_ceiling_fan_start_dimmer`
- `button.bedroom_ceiling_fan_stop_actions`

The first one starts the dimming or brightening the light -- it cycles back and forth and will continue the cycle indefinately until the second `button` is selected.

So, on the NSPanal I have Button 1 set to toggle that light on and off, and I'd like to add:
- Press and Hold on the button activates:  `button.bedroom_ceiling_fan_start_dimmer`
- Release on the Held Button activates: `button.bedroom_ceiling_fan_stop_actions`

Any suggestions?

Thanks.

**Comment by edwardtfn** [comment]
The button state is exposed to Home Assistant as a binary sensor independently if you assigned anything to the button, so you could do this as a separated automation on HA side.
